app/controllers/home_controller.rb

```diff
@@ -204,7 +204,63 @@ def recent_activity
 
   end
 
+
+  #
+  # Valid set of stem finder filter names and values.
+  #
+  @@STEM_FILTER_VALIDATION =
+    {   "subject"       =>  {
+                                "physics-chemistry" => {},
+                                "life-sciences"     => {},
+                                "engineering-tech"  => {},
+                                "earth-space"       => {},
+                                "mathematics"       => {}
+                            },
+        "grade-level"   =>  {
+                                "elementary-school" => {},
+                                "middle-school"     => {},
+                                "high-school"       => {},
+                                "higher-education"  => {}
+                            } 
+    }
+
+
+  #
+  #
+  # Handle /stem-reources/ routes to either pre-populate stem finder filters
+  # or render an individual resource lightbox.
+  #
+  #
   def stem_resources
+
+    # logger.info("INFO stem_resources called.")
+
+    #
+    # See if we are filtering categories in URL to pass to
+    # stem finder.
+    #
+    if params[:filter_name] && params[:filter_value]
+        filter_name     = params[:filter_name]
+        filter_value    = params[:filter_value]
+
+        # logger.info("INFO filter_name/filter_value #{filter_name} #{filter_value}.")
+
+        if  @@STEM_FILTER_VALIDATION.key?(filter_name) &&
+            @@STEM_FILTER_VALIDATION[filter_name].key?(filter_value)
```

Comment:
In order to give the home page the most flexibility on constructing its URLs without needing to change the portal code.  I'd suggest that we send everything to the index page if it doesn't have a param[:type] of activity or sequence.  This does mean that this action will only send a 404 if the type is activity or sequence and the id doesn't exist.  So it will be up to the home page to show an appropriate error for invalid URLs.  But I think that is a good comprise so we don't need to maintain the URL rules in two different codebases.